### PR TITLE
Edit basket

### PR DESCRIPTION
**Pull request by Sajaas**
## PT: https://www.pivotaltracker.com/story/show/152642684 ##
* User can clear cart
* User can remove item

### CODE REVIEW

File: e2e/app.e2e-spec.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ describe('kimchistan-front-end App', () => {
     browser.wait(function () {
       return element(by.id('dropdown_Bibimbap')).isPresent();
     }, 5000);
-    element(by.id('dropdown_Bibimbap')).click();
+    element(by.id(`dropdown_Bibimbap`)).click();
```
Comment (tochman) [inline e2e/app.e2e-spec.ts]: Why did you change to back-ticks?

File: src/app/service/shoppingcart.service.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { Injectable } from '@angular/core';
+import {Injectable} from '@angular/core';
```
Comment (tochman) [inline src/app/service/shoppingcart.service.ts]: I would like you to agree to a standard for this (`{ Injectable }` or `{Injectable} `) and be consistent throughout the project.

File: src/app/service/shoppingcart.service.ts [hunk 3]
```diff
@@ -53,4 +46,29 @@ export class ShoppingCartService {
   loadCart() {
     this.cart = JSON.parse(localStorage.getItem('cart'));
   }
+
+  clearCart() {
+    this.cart = [];
+    this.saveCart();
+  }
+
+  removeProduct(p_id: string, p_name: string, price: number, i_id: string, i_name: string) {
```
Comment (tochman) [inline src/app/service/shoppingcart.service.ts]: sorry, but the argument names are confusing

File: e2e/app.e2e-spec.ts (hunk 1 above)
Comment (Sajaas) [reply e2e/app.e2e-spec.ts]: Probably some sneaky Icelandic keyboard ;)

Comment (tochman) [reply e2e/app.e2e-spec.ts]: This COULD be a problem though...

Comment (Sajaas) [reply e2e/app.e2e-spec.ts]: Yes, I will change it today, as well as acting on the rest of your comments. :)

File: src/app/service/shoppingcart.service.ts (hunk 3 above)
Comment (Sajaas) [reply src/app/service/shoppingcart.service.ts]: Will check it out

File: src/app/app.component.html [hunk 4]
```diff
@@ -11,13 +11,24 @@
               <div *ngFor="let item of cart">
                 <div *ngIf="item.ingredient_name">
                   {{item.product_name}} med {{item.ingredient_name}} ({{item.price}} kr) x{{item.item_quantity}}
+                  <button class="btn-remove" id="remove_{{item.product_name}}-{{item.ingredient_name}}"
+                          (click)="removeProduct(item.product_id, item.product_name, item.price, item.ingredient_id, item.ingredient_name)">
```
Comment (diraulo) [inline src/app/app.component.html]: Why not pass the `item` only then as an argument then extract all those fields within the remove product method? 🤔

File: src/app/app.component.html [hunk 5]
```diff
@@ -11,13 +11,24 @@
...
+                    Ta bort
+                  </button>
                 </div>
                 <div *ngIf="item.ingredient_name == undefined">
                   {{item.product_name}} ({{item.price}} kr) x{{item.item_quantity}}
+                  <button class="btn-remove" id="remove{{item.product_name}}"
+                          (click)="removeProduct(item.product_id, item.product_name, item.price)">
```
Comment (diraulo) [inline src/app/app.component.html:22]: same here just send in the `item`?

**Review by diraulo — COMMENTED** [review]
see comments...

File: src/app/app.component.html (hunk 5 above)
Comment (magnus-thor) [reply src/app/app.component.html:22]: We will do this if/when we have time for some refactoring.  We started on it and it  starting breaking the tests and we didn't want to spend any more time on it, since have other stuff that needs to be done to get this up and working.
